File: 70 climbing-stairs/c/main.v2.c
/**
 * Leetcode最优解
 */

int climbStairs(int n) {
    if(n == 1) {
      return 1;
    }

    if(n == 2) {
      return 2;
    }

    int arr[n+1];
    arr[0] = 0;
    arr[1] = 1;
    arr[2] = 2;

    int i;
    for(i = 3; i <= n; i++) {
        arr[i] = arr[i-1] + arr[i-2];
    }

    return arr[n];
  }

File: 637 average-of-levels-in-binary-tree/js/main.v2.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

/**
 * Leetcode最优解
 * 使用循环，替代迭代的方法
 */

function pushLeavesToArray(arr, node) {
  if (node.left) {
    arr.push(node.left)
  }
  if (node.right) {
    arr.push(node.right)
  }
  return arr
}

var averageOfLevels = function (root) {
  var results = []
  results.push(root.val)
  var nextLevel = []

  pushLeavesToArray(nextLevel, root)
  while (nextLevel.length > 0) {
    var tmpLevel = []
    var result = 0
    nextLevel.forEach(function (n) {
      tmpLevel = pushLeavesToArray(tmpLevel, n)
      result += n.val
    })
    results.push(result / nextLevel.length)
    nextLevel = tmpLevel
  }
  return results
};
